zebra/connected.c

```diff
@@ -210,10 +211,14 @@ void connected_up(struct interface *ifp, struct connected *ifc)
 	UNSET_FLAG(ifc->conf, ZEBRA_IFC_DOWN);
 
 	prefix_copy(&p, CONNECTED_PREFIX(ifc));
+	prefix_copy(&plocal, CONNECTED_PREFIX(ifc));
 
 	/* Apply mask to the network. */
 	apply_mask(&p);
 
+	if (prefix_same(&p, &plocal))
+		install_local = false;
```

Comment:
Hmm. This creates inconsistency if you're trying to advertise/redistribute all your local addresses, because you need to special-case /32s and /128s. Is there a problem if we have identical `C>` and `L>` for this case?

Also it's doing the wrong thing if you configure a `::0` address in IPv6 (which is weird but possible), e.g. `ipv6 address 2001:db8::/64` on an interface. It would need to be below the prefix length change below. But it's moot if we remove the check altogether.

Reply:
I was trying to avoid this display:
```
L * 27.0.0.16/32 is directly connected, lo, 00:00:30
C>* 27.0.0.16/32 is directly connected, lo, 00:00:30
```
as that they were the same.  Not sure there is a problem.

Reply:
fixed